lib/usart/src/stm32h7.rs

```diff
@@ -0,0 +1,149 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! STM32H7-specific USART details.
+
+pub use drv_stm32xx_sys_api::Sys;
+
+use super::BaudRate;
+
+#[cfg(feature = "stm32h743")]
+use stm32h7::stm32h743 as device;
+
+#[cfg(feature = "stm32h753")]
+use stm32h7::stm32h753 as device;
+
+/// Which USART device
+#[derive(Debug, Clone, Copy)]
+pub enum DeviceId {
+    Usart3,
+}
+
+pub struct Device(&'static device::usart1::RegisterBlock);
+
+impl Device {
+    /// Turn on the `device` USART with the given baud rate.
+    // TODO passing in a `Sys` seems weird. Should we take the sys task ID
+    // instead?  Something else?
+    pub fn turn_on(
+        sys: &Sys,
+        device: DeviceId,
+        clock_hz: u32,
+        baud_rate: BaudRate,
+    ) -> Self {
+        // Turn the actual peripheral on so that we can interact with it.
+        turn_on_usart(sys, device);
+
+        let usart = match device {
+            DeviceId::Usart3 => {
+                // From thin air, pluck a pointer to the USART register block.
+                //
+                // Safety: this is needlessly unsafe in the API. The USART is
+                // essentially a static, and we access it through a & reference
+                // so aliasing is not a concern. Were it literally a static, we
+                // could just reference it.
+                #[cfg(any(feature = "stm32h743", feature = "stm32h753"))]
```

Comment:
(this cfg should probably get hoisted higher, since if it fails there'll be a weird error message about `()`. Or, removed!)